### PR TITLE
Add support to install linux OS using vm_mac paramter

### PR DESCRIPTION
**Pull request by spoorthys1303**
Installation of Linux OS required changes in lpar_netboot command with addition to input paramter vm_mac from user.
When user inputs the vm_mac paramter, linux os installtion starts with the the help of lpar_netboot command with the additional options and vm_mac.

### CODE REVIEW

File: plugins/module_utils/hmc_resource.py [hunk 1]
```diff
@@ -482,21 +482,53 @@ def fetchIODetailsForNetboot(self, nimIP, gateway, lparIP, viosName, profName, s
...
+        cmd = 'mkvterm -m ' + systemName + ' -p ' + viosName
+        stdout = conn.execute(cmd)
+        try:
+            for line in stdout:
+                logger.debug(line.strip())
+                logger.debug("\n")
+        except UnicodeDecodeError:
+            pass
+
+    def checkconsolelog(self, module, lpar_ip, lpar_hmc, userid, hmc_password, systemName, viosName):
```
Comment (naveenkandakur) [inline plugins/module_utils/hmc_resource.py]: Are we passing vios_name here?, if no please change the parameter name to lpar_name

File: plugins/modules/powervm_lpar_instance.py [hunk 2]
```diff
@@ -1512,6 +1535,9 @@ def install_aix_os(module, params):
 
         if location_code:
             hmc.installOSFromNIM(location_code, nim_ip, nim_gateway, vm_ip, nim_vlan_id, nim_vlan_priority, nim_subnetmask, vm_name, profile_name, system_name)
+        elif vm_mac:
+            hmc.installOSFromNIM(location_code, nim_ip, nim_gateway, vm_ip, None, None, nim_subnetmask, vm_name, profile_name, system_name, vm_mac)
+            hmc.checkconsolelog(module, vm_ip, hmc_host, hmc_user, password, system_name, vm_name)
```
Comment (naveenkandakur) [inline plugins/modules/powervm_lpar_instance.py:1541]: Is this method valid only when user passes vm_mac? If no please enable the same in other places too like when used pass the loc_code and doesn't pass the loc_code and in VIOS module too

**Review by naveenkandakur — CHANGES_REQUESTED** [review]

File: plugins/module_utils/hmc_resource.py (hunk 1 above)
Comment (spoorthys1303) [reply plugins/module_utils/hmc_resource.py]: changed to lparName

File: plugins/modules/powervm_lpar_instance.py (hunk 2 above)
Comment (spoorthys1303) [reply plugins/modules/powervm_lpar_instance.py:1541]: yes when user pass vm_mac, check_console will run

**Comment by spoorthys1303** [comment]
Added instruction to suggest user to use location_code for AIX installation and vm_mac for Linux installation

**Review by naveenkandakur — APPROVED** [review]

**Pull request merged**